npm/test/filecheckprocessing.test.ts

```diff
@@ -0,0 +1,140 @@
+`(
+  "updateProgressBar sets the progress bar value to $expected for processed $processed and total $total",
+  ({ processed, total, expected }) => {
+    document.body.innerHTML = `<progress id="test" value="0" max="100"/>`
+    updateProgressBar(processed, total, "#test")
+    const progressBar: HTMLProgressElement | null = document.querySelector(
+      "#test"
+    )
+    expect(progressBar).not.toBeNull()
+    expect(progressBar!.value).toBe(expected)
+  }
+)
+
+test("getConsignmentData returns the correct consignment data with a successful api call", done => {
+  const consignmentId = "7d4ae1dd-caeb-496d-b503-ab0e8d82a12c"
+  const clientMock = GraphqlClient as jest.Mock
+  clientMock.mockImplementation(() => new GraphqlClientSuccess())
+  const client = new GraphqlClient("https://test.im", mockKeycloakInstance)
+  document.body.innerHTML = `<input id="consignmentId" type="hidden" value="${consignmentId}">`
+
+  const callback: (
+    fileCheckProcessed: IFileCheckProcessed | null
+  ) => void = fileCheckProcessed => {
+    expect(fileCheckProcessed!.antivirusProcessed).toBe(2)
+    expect(fileCheckProcessed!.checksumProcessed).toBe(3)
+    expect(fileCheckProcessed!.totalFiles).toBe(10)
+    done()
+  }
+  getConsignmentData(client, callback)
+})
+
+test("getConsignmentData returns null with a failed api call", done => {
+  const consignmentId = "7d4ae1dd-caeb-496d-b503-ab0e8d82a12c"
+  const clientMock = GraphqlClient as jest.Mock
+  clientMock.mockImplementation(() => new GraphqlClientFailure())
+  const client = new GraphqlClient("https://test.im", mockKeycloakInstance)
+  document.body.innerHTML = `<input id="consignmentId" type="hidden" value="${consignmentId}">`
+
+  const callback: (
+    fileCheckProcessed: IFileCheckProcessed | null
+  ) => void = fileCheckProcessed => {
+    expect(fileCheckProcessed).toBeNull()
+    done()
+  }
+  getConsignmentData(client, callback)
+})
+
+test("getConsignmentData returns null with empty data from a successful api call", done => {
+  const consignmentId = "7d4ae1dd-caeb-496d-b503-ab0e8d82a12c"
+  const clientMock = GraphqlClient as jest.Mock
+  clientMock.mockImplementation(() => new GraphqlClientEmptyData())
+  const client = new GraphqlClient("https://test.im", mockKeycloakInstance)
+  document.body.innerHTML = `<input id="consignmentId" type="hidden" value="${consignmentId}">`
+
+  const callback: (
+    fileCheckProcessed: IFileCheckProcessed | null
+  ) => void = fileCheckProcessed => {
+    expect(fileCheckProcessed).toBeNull()
+    done()
+  }
+  getConsignmentData(client, callback)
+})
+
+test("getConsignmentId returns the correct id when the hidden input is present", () => {
+  const consignmentId = "91847ae4-3bfd-4fac-b448-68948dd95820"
+  document.body.innerHTML = `<input id="consignmentId" type="hidden" value="${consignmentId}">`
+  expect(getConsignmentId()).toBe(consignmentId)
+})
+
+test("getConsignmentId returns the correct id when the hidden input is absent", () => {
```

Comment:
Should this be about returning an error rather than the correct id?

Reply:
Yep, good spot. Looks like a bad copy paste job. I've changed it.